apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.html.ts

```diff
@@ -0,0 +1,557 @@
+        name="add_member_uids"
+      />
+      <br />
+      <input type="hidden" name="__action" value="add_member" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+      <input type="hidden" name="group_id" value="${row.group_id}" />
+      <button
+        type="button"
+        class="btn btn-secondary"
+        onclick="$('#row${row.group_id}PopoverAdd').popover('hide')"
+      >
+        Cancel
+      </button>
+      <button type="submit" class="btn btn-primary">Add</button>
+    </form>
+  `;
+}
+
+function DeleteGroupForm({ row, csrfToken }: { row: GroupUsersRow; csrfToken: string }) {
+  return html`
+    <form name="delete-group-form" method="POST">
+      <p>Are you sure you want to delete the group <strong>${row.name}</strong>?</p>
+      <input type="hidden" name="__action" value="delete_group" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+      <input type="hidden" name="group_id" value="${row.group_id}" />
+      <button
+        type="button"
+        class="btn btn-secondary"
+        onclick="$('#row${row.group_id}Popoverdeletegroup').popover('hide')"
+      >
+        Cancel
+      </button>
+      <button type="submit" class="btn btn-danger">Delete</button>
+    </form>
+  `;
+}
+
+function RemoveMembersForm({ row, csrfToken }: { row: GroupUsersRow; csrfToken: string }) {
+  return html`
+    <form name="delete-member-form" method="POST">
+      <div class="form-group">
+        <label for="delete-member-form-${row.group_id}">UID:</label>
+        <select class="form-control" name="user_id" id="delete-member-form-${row.group_id}">
+          ${row.users.map((user) => {
+            return html` <option value="${user.user_id}">${user.uid}</option> `;
+          })}
+        </select>
+      </div>
+      <input type="hidden" name="__action" value="delete_member" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+      <input type="hidden" name="group_id" value="${row.group_id}" />
+      <button
+        type="button"
+        class="btn btn-secondary"
+        onclick="$('#row${row.group_id}Popoverdeletemember').popover('hide')"
+      >
+        Cancel
+      </button>
+      <button type="submit" class="btn btn-danger" ${row.users.length > 0 ? '' : 'disabled'}>
+        Delete
+      </button>
+    </form>
+  `;
+}
+
+function UploadAssessmentGroupsModal({ csrfToken }: { csrfToken: string }) {
+  return Modal({
+    id: 'uploadAssessmentGroupsModal',
+    title: 'Upload new group assignments',
+    formEncType: 'multipart/form-data',
+    body: html`
+      <p>Upload a CSV file in the format of:</p>
+      <code class="mb-0 text-dark">
+        groupName,UID<br />
+        groupA,one@example.com<br />
+        groupA,two@example.com<br />
+        groupB,three@example.com<br />
+        groupB,four@example.com</code
+      >
+      <!-- Closing code tag not on its own line to improve copy/paste formatting -->
+      </div>
```

Comment:
What does this closing `</div>` correspond to?

Reply:
Nothing, as the opening `<div>` was for 'modal-body'. Removed this closing `</div>` in [49b24ad](https://github.com/PrairieLearn/PrairieLearn/pull/9550/commits/49b24ad27a42ff43993f2210f2ac2a3d611c3f11).